File: app/src/main/java/com/example/tetrisapp/room/model/LeaderboardEntry.java
package com.example.tetrisapp.room.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

import com.example.tetrisapp.room.converter.DateConverter;

import java.util.Date;

@Entity
@TypeConverters({DateConverter.class})
public class LeaderboardEntry {
    @PrimaryKey(autoGenerate = true)
    public long id = 0;

    @ColumnInfo(name = "score")
    public int score;

    @ColumnInfo(name = "level")
    public int level;

    @ColumnInfo(name = "lines")
    public int lines;

    @ColumnInfo(name = "date")
    public Date date;
}

File: app/src/main/java/com/example/tetrisapp/model/local/LeaderboardEntry.kt
package com.example.tetrisapp.model.local

import androidx.room.*
import com.example.tetrisapp.data.local.converter.DateConverter
import java.util.*

@Entity(
    indices = [Index(value = ["hash"], unique = true)]
)
@TypeConverters(DateConverter::class)
class LeaderboardEntry {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    @ColumnInfo(name = "score", defaultValue = "0")
    var score = 0

    @ColumnInfo(name = "level", defaultValue = "0")
    var level = 0

    @ColumnInfo(name = "lines", defaultValue = "0")
    var lines = 0

    @ColumnInfo(name = "date")
    var date: Date? = null

    @ColumnInfo(name = "timeInGame", defaultValue = "0")
    var timeInGame = 0

    @ColumnInfo(name = "uploaded", defaultValue = "false")
    var uploaded = false

    @ColumnInfo(name = "hash")
    var hash: String? = null
}